corgi/core/models.py

```diff
@@ -819,27 +825,52 @@ class Meta:
             models.Index(fields=["product_streams"]),
             models.Index(fields=["product_variants"]),
             models.Index(fields=("type", "product_streams")),
+            models.Index(
+                fields=("type", "name"),
```

Comment:
Not sure if below is needed, but we do want to filter on it...
```suggestion
                fields=("type", "name", "arch"),
```